### PR TITLE
KAA-761: Remove deprecated API to create singleton instance of C++ SDK.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-cpp/impl/channel/connectivity/PingConnectivityChecker.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/connect.hpp>
+#include <boost/asio/io_service.hpp>
+
+#include "kaa/logging/Log.hpp"
+
+namespace kaa {
+
+const std::string PingConnectivityChecker::DEFAULT_HOST = "www.google.com";
```
Comment (Karnil) [inline client/client-multi/client-cpp/impl/channel/connectivity/PingConnectivityChecker.cpp:31]: why are you using "google.com" as the default host? what would happen if that host is not available?

Comment (ghost) [inline client/client-multi/client-cpp/impl/channel/connectivity/PingConnectivityChecker.cpp:31]: It was made only for compatibility with Java SDK.
Insied C++ SDK uses its own connectivity checker than pings one of Kaa Bootstrap server.
See details in https://github.com/kaaproject/kaa/blob/master/client/client-multi/client-cpp/impl/channel/connectivity/IPConnectivityChecker.cpp

File: client/client-multi/client-cpp/kaa/KaaThread.hpp [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 #include "kaa/KaaDefaults.hpp"
 
-#ifdef KAA_THREADSAFE
+//#ifdef KAA_THREADSAFE
```
Comment (Karnil) [inline client/client-multi/client-cpp/kaa/KaaThread.hpp:22]: Maybe it should be removed?

File: client/client-multi/client-cpp/kaa/KaaThread.hpp [hunk 3]
```diff
@@ -57,7 +57,7 @@ typedef std::atomic_int_fast32_t RequestId;
 
 #define kaa_thread_local thread_local
 
-#else
+#if 0
```
Comment (Karnil) [inline client/client-multi/client-cpp/kaa/KaaThread.hpp:60]: You should to remove this block.

Comment (ghost) [inline client/client-multi/client-cpp/kaa/KaaThread.hpp:60]: I agree, but let me do this in other task.

File: client/client-multi/client-cpp/kaa/KaaThread.hpp (hunk 2 above)
Comment (ghost) [inline client/client-multi/client-cpp/kaa/KaaThread.hpp:22]: I agree, but let me do this in other task.

**Pull request merged**